### PR TITLE
:warning: Rename tagtype to tagcategory

### PR DESCRIPTION
**Pull request by gildub**
Resolves https://github.com/konveyor/tackle2-ui/issues/665

### CODE REVIEW

File: client/public/locales/en/translation.json [hunk 1]
```diff
@@ -97,14 +97,14 @@
       "newStakeholder": "New stakeholder",
       "newStakeholderGroup": "New stakeholder group",
       "newTag": "New Tag",
-      "newTagType": "New tag type",
+      "newtagcategory": "New tag category",
```
Comment (ibolton336) [inline client/public/locales/en/translation.json:100]: @gildub Shouldn't this be camel case?

File: client/public/locales/en/translation.json [hunk 2]
```diff
@@ -97,14 +97,14 @@
...
       "updateApplication": "Update application",
       "updateBusinessService": "Update business service",
       "updateJobFunction": "Update job function",
       "updateStakeholder": "Update stakeholder",
       "updateStakeholderGroup": "Update stakeholder group",
       "updateTag": "Update tag",
-      "updateTagType": "Update tag type"
+      "updatetagcategory": "Update tag category"
```
Comment (ibolton336) [inline client/public/locales/en/translation.json:107]: Camel case?

File: client/public/locales/en/translation.json [hunk 3]
```diff
@@ -286,12 +286,12 @@
...
     "tagCount": "Tag count",
-    "tagDeleted": "",
+    "tagDelete": "Tag deleted",
     "tagName": "Tag name",
     "tags": "Tags",
-    "tagType": "Tag type",
-    "tagTypeDeleted": "",
-    "tagTypes": "Tag types",
+    "tagcategory": "Tag category",
+    "tagcategoryDelete": "Tag category deleted",
```
Comment (ibolton336) [inline client/public/locales/en/translation.json:293]: Camel case?

File: client/public/locales/es/translation.json [hunk 4]
```diff
@@ -287,9 +287,9 @@
     "tagDeleted": "Etiqueta descheda",
     "tagName": "Nombre de etiqueta",
     "tags": "Etiquetas",
-    "tagType": "Tipo de etiqueta",
-    "tagTypeDeleted": "Tipo de etiqueta descheda",
-    "tagTypes": "Tipos de etiqueta",
+    "tagcategory": "Categoría de etiqueta",
+    "tagcategoryDeleted": "Categoría de etiqueta descheda",
+    "tagcategories": "Categorías de etiqueta",
```
Comment (ibolton336) [inline client/public/locales/es/translation.json:292]: CC?

File: client/src/app/pages/applications/components/application-tags/application-tags.tsx [hunk 5]
```diff
@@ -20,8 +20,10 @@ export interface ApplicationTagsProps {
 export const ApplicationTags: React.FC<ApplicationTagsProps> = ({
   application,
 }) => {
-  const [tagTypes, setTagTypes] = useState<Map<number, TagType>>(new Map()); // <tagTypeId, tagType>
-  const [tags, setTags] = useState<Map<number, Tag[]>>(new Map()); // <tagTypeId, tags[]>
+  const [tagCategories, setTagCategories] = useState<Map<number, TagCategory>>(
+    new Map()
+  ); // <tagCategoryId, tagCategory>
```
Comment (ibolton336) [inline client/src/app/pages/applications/components/application-tags/application-tags.tsx:25]: Comment removal suggestion

File: client/src/app/pages/applications/components/application-tags/application-tags.tsx [hunk 6]
```diff
@@ -48,38 +50,44 @@ export const ApplicationTags: React.FC<ApplicationTagsProps> = ({
 
           Promise.all(
             tagValidResponses.map((tag) =>
-              getTagTypeById(tag?.tagType?.id || 0)
+              getTagCategoryById(tag?.category?.id || 0)
```
Comment (ibolton336) [inline client/src/app/pages/applications/components/application-tags/application-tags.tsx:53]: so this is .category and not .tagCategory? Just a question.

File: client/src/app/pages/controls/tags/components/new-tag-type-modal/index.ts [hunk 7]
```diff
@@ -1 +1 @@
-export { NewTagTypeModal } from "./new-tag-type-modal";
+export { NewTagCategoryModal as NewTagCategoryModal } from "./new-tag-type-modal";
```
Comment (ibolton336) [inline client/src/app/pages/controls/tags/components/new-tag-type-modal/index.ts:1]: What does this do?

File: client/src/app/pages/controls/tags/components/new-tag-type-modal/new-tag-type-modal.tsx [hunk 8]
```diff
@@ -23,12 +23,12 @@ export const NewTagTypeModal: React.FC<NewTagTypeModalProps> = ({
 
   return (
     <Modal
-      title={t("dialog.title.newTagType")}
+      title={t("dialog.title.newtagcategory")}
```
Comment (ibolton336) [inline client/src/app/pages/controls/tags/components/new-tag-type-modal/new-tag-type-modal.tsx:26]: Camel case ?

File: client/src/app/pages/controls/tags/components/tag-form/tag-form.tsx [hunk 9]
```diff
@@ -158,14 +160,14 @@ export const TagForm: React.FC<TagFormProps> = ({ tag, onSaved, onCancel }) => {
           />
         </FormGroup>
         <FormGroup
-          label={t("terms.tagType")}
-          fieldId="tagType"
+          label={t("terms.tagcategory")}
```
Comment (ibolton336) [inline client/src/app/pages/controls/tags/components/tag-form/tag-form.tsx:163]: Camel case?

File: client/src/app/pages/controls/tags/components/tag-form/tag-form.tsx [hunk 10]
```diff
@@ -176,15 +178,15 @@ export const TagForm: React.FC<TagFormProps> = ({ tag, onSaved, onCancel }) => {
               clearSelectionsAriaLabel: "tag-type",
               removeSelectionAriaLabel: "tag-type",
               placeholderText: t("composed.selectOne", {
-                what: t("terms.tagType").toLowerCase(),
+                what: t("terms.tagcategory").toLowerCase(),
```
Comment (ibolton336) [inline client/src/app/pages/controls/tags/components/tag-form/tag-form.tsx:181]: Camel Case?

File: client/src/app/pages/controls/tags/components/update-tag-category-modal/update-tag-category-modal.tsx [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+export const UpdateTagCategoryModal: React.FC<UpdateTagCategoryModalProps> = ({
+  tagCategory: tagCategory,
+  onSaved,
+  onCancel,
+}) => {
+  const { t } = useTranslation();
+
+  return (
+    <Modal
+      title={t("dialog.title.updatetagcategory")}
```
Comment (ibolton336) [inline client/src/app/pages/controls/tags/components/update-tag-category-modal/update-tag-category-modal.tsx:26]: Camel Case?

File: client/src/app/pages/controls/tags/tags.tsx [hunk 12]
```diff
@@ -110,15 +112,15 @@ export const Tags: React.FC = () => {
     onDeleteTagError
   );
 
-  const onDeleteTagTypeSuccess = (response: any) => {
+  const onDeleteTagCategorySuccess = (response: any) => {
     pushNotification({
-      title: t("terms.tagTypeDeleted"),
+      title: t("terms.tagcategoryDeleted"),
```
Comment (ibolton336) [inline client/src/app/pages/controls/tags/tags.tsx:117]: Camel Case?

File: client/src/app/pages/controls/tags/tags.tsx [hunk 13]
```diff
@@ -244,7 +251,7 @@ export const Tags: React.FC = () => {
 
   const columns: ICell[] = [
     {
-      title: t("terms.tagType"),
+      title: t("terms.tagcategory"),
```
Comment (ibolton336) [inline client/src/app/pages/controls/tags/tags.tsx:254]: Camel Case?

File: client/src/app/pages/controls/tags/tags.tsx [hunk 14]
```diff
@@ -447,39 +456,39 @@ export const Tags: React.FC = () => {
...
+                    {t("actions.createTagCategory")}
                   </Button>
                 </ToolbarItem>
               </RBAC>
             </ToolbarGroup>
           }
           noDataState={
             <NoDataEmptyState
-              // t('terms.tagTypes')
+              // t('terms.tagcategories')
```
Comment (ibolton336) [inline client/src/app/pages/controls/tags/tags.tsx:469]: Comment removal suggestion

File: client/src/app/pages/controls/tags/tags.tsx [hunk 15]
```diff
@@ -447,39 +456,39 @@ export const Tags: React.FC = () => {
...
               title={t("composed.noDataStateTitle", {
-                what: t("terms.tagTypes").toLowerCase(),
+                what: t("terms.tagcategories").toLowerCase(),
```
Comment (ibolton336) [inline client/src/app/pages/controls/tags/tags.tsx:471]: Camel Case?

File: client/src/app/pages/controls/tags/tags.tsx [hunk 16]
```diff
@@ -447,39 +456,39 @@ export const Tags: React.FC = () => {
...
               })}
               // t('terms.stakeholderGroup')
               description={
                 t("composed.noDataStateBody", {
-                  what: t("terms.tagType").toLowerCase(),
+                  what: t("terms.tagcategory").toLowerCase(),
```
Comment (ibolton336) [inline client/src/app/pages/controls/tags/tags.tsx:476]: Camel Case?

File: client/src/app/pages/controls/tags/tags.tsx [hunk 17]
```diff
@@ -515,25 +524,25 @@ export const Tags: React.FC = () => {
           }}
         />
       )}
-      {isTagTypeToDeleteConfirmDialogOpen && (
+      {isTagCategoryToDeleteConfirmDialogOpen && (
         <ConfirmDialog
           title={t("dialog.title.delete", {
-            what: t("terms.tagType").toLowerCase(),
+            what: t("terms.tagcategory").toLowerCase(),
```
Comment (ibolton336) [inline client/src/app/pages/controls/tags/tags.tsx:530]: Camel Case?

File: client/src/app/queries/tags.ts [hunk 18]
```diff
@@ -11,15 +16,15 @@ export interface ITagFetchState {
...
+export interface ITagCategoryFetchState {
+  tagCategories: TagCategory[];
   isFetching: boolean;
   fetchError: any;
   refetch: any;
 }
 
 export const TagsQueryKey = "tags";
-export const TagTypesQueryKey = "tagtypes";
+export const TagCategoriesQueryKey = "tagcategory";
```
Comment (ibolton336) [inline client/src/app/queries/tags.ts:27]: Camel Case?

File: client/src/app/utils/model-utils.tsx [hunk 19]
```diff
@@ -115,30 +115,33 @@ export const toIJobFunctionDropdownOptionWithValue = (
...
+  value: ITagCategoryDropdown
+): OptionWithValue<ITagCategoryDropdown> => ({
   value,
   toString: () => value.name,
 });
 
 // Tag
 
-export interface ITagDropdown extends Pick<Tag, "id" | "name" | "tagType"> {}
+export interface ITagDropdown extends Pick<Tag, "id" | "name" | "category"> {}
```
Comment (ibolton336) [inline client/src/app/utils/model-utils.tsx:139]: Why not tagCategory?

**Review by ibolton336 — CHANGES_REQUESTED** [review]

File: client/src/app/utils/model-utils.tsx (hunk 19 above)
Comment (gildub) [reply client/src/app/utils/model-utils.tsx:139]: Because the backend has changed the Tag model.

File: client/src/app/queries/tags.ts (hunk 18 above)
Comment (gildub) [reply client/src/app/queries/tags.ts:27]: Not for the fetch query key.

File: client/src/app/pages/applications/components/application-tags/application-tags.tsx (hunk 6 above)
Comment (gildub) [reply client/src/app/pages/applications/components/application-tags/application-tags.tsx:53]: Yes the backend changed tag.tagCategory to tag.category

File: client/src/app/pages/controls/tags/components/new-tag-type-modal/index.ts (hunk 7 above)
Comment (gildub) [reply client/src/app/pages/controls/tags/components/new-tag-type-modal/index.ts:1]: That's vscode side effect !

**Comment by ibolton336** [comment]
@gildub Thanks for doing this!

**Review by ibolton336 — APPROVED** [review]

**Pull request merged**